### DISCUSSION TITLE
support local setup for test purposes

### DISCUSSION THREAD

**Discussion by orltom** [fluxcd/flux2#4104] [Proposals]
As a developer, you want to be able to map the productive environment locally. In our case, we use Flux to provision components and map their dependencies and health checks. This is very helpful for us. If you want to keep the drift from the production setup as small as possible, it would be beneficial to have FluxCD support the local filesystem as SourceRef and be able to update the filesystem on change. I know this does not reflect the GitOps approach, but it would help many developers to have such a feature. What do you think or do you have an alternative suggestion?

To give more context, we use k3d locally to emulate the Kubernetes Platform and Tilt to apply Kubernetes Manifest/Helm Charts. Currently, we map the dependencies and health checks from the Kustomize CRs via Tilt, which is double-mobbed and the same behavior cannot always be achieved.

**Comment by hiddeco** [comment]
See https://github.com/stefanprodan/flux-local-dev on how to achieve this through with OCI and a local registry.
